superset/common/query_object.py

```diff
@@ -254,6 +264,14 @@ def validate(
         """Validate query object"""
         error: Optional[QueryObjectValidationError] = None
         all_labels = self.metric_names + self.column_names
+        missing_series = [col for col in self.series_columns if col not in self.columns]
+        if missing_series:
+            _(
```

Comment:
@villebro you forgot to raise the exception here 

https://github.com/apache/superset/pull/16946 fix it